### PR TITLE
Archie/dwt/popup refactor

### PR DESCRIPTION
**Pull request by archiepeterjarvis**
Beautify/refactor the DWT validation popup.

image

### CODE REVIEW

File: src/asmcnc/comms/localization.py [hunk 1]
```diff
@@ -22,16 +23,17 @@
...
-
 builder_font_string = """
 <Widget>:
     font_name: "%s"
     title_font: "%s"
 """
 
 
-
 class Localization(object):
```
Comment (Masc999) [inline src/asmcnc/comms/localization.py:33]: Please add a doc string for the class. Especially as it is a singlteton now.

File: src/asmcnc/core_UI/new_popups/job_validation_popup.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from kivy.uix.boxlayout import BoxLayout
+from kivy.uix.button import Button
+from kivy.uix.label import Label
+
+from asmcnc.comms.localization import Localization
+from asmcnc.core_UI import scaling_utils
+from asmcnc.core_UI.new_popups.popup_bases import PopupBase, PopupErrorTitle, PopupScrollableBody
+
+
+class JobValidationPopup(PopupBase):
```
Comment (Masc999) [inline src/asmcnc/core_UI/new_popups/job_validation_popup.py:11]: Add docstring and decribe what this is used for and how to use it.

File: src/asmcnc/core_UI/new_popups/job_validation_popup.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    def __init__(self, text, **kwargs):
+        super(JobValidationPopup, self).__init__(**kwargs)
+        localisation = Localization()
+
+        title = PopupErrorTitle(size_hint_y=0.15, localisation=localisation)
```
Comment (Masc999) [inline src/asmcnc/core_UI/new_popups/job_validation_popup.py:16]: Why not use localization as singleton directly in the class and not forwarding it?

File: src/asmcnc/comms/localization.py [hunk 4]
```diff
@@ -74,14 +76,23 @@ class Localization(object):
     ORIGINAL_PRODUCT_NAME = "SmartBench"
     PRODUCT_NAME = "SmartBench"
 
+    def __new__(cls):
```
Comment (Masc999) [inline src/asmcnc/comms/localization.py:79]: 👍

File: src/asmcnc/core_UI/new_popups/job_validation_popup.py (hunk 3 above)
Comment (archiepeterjarvis) [reply src/asmcnc/core_UI/new_popups/job_validation_popup.py:16]: It is always instantiated from a class which will have localisation. Acquiring the lock is an expensive operation so I thought this makes sense.

Comment (archiepeterjarvis) [reply src/asmcnc/core_UI/new_popups/job_validation_popup.py:16]: But if you think otherwise happy to change it 😁

Comment (Masc999) [reply src/asmcnc/core_UI/new_popups/job_validation_popup.py:16]: 👍

**Pull request merged**